csharp/device/Microsoft.Azure.Devices.Client/HttpClientHelper.cs

```diff
@@ -59,7 +90,50 @@ sealed class HttpClientHelper : IHttpClientHelper
         {
             return this.GetAsync<T>(requestUri, errorMappingOverrides, customHeaders, true, cancellationToken);
         }
+#endif
+
+#if NETMF
+        public HttpWebResponse Get(
+            string requestUri,
+            Hashtable customHeaders,
+            bool throwIfNotFound)
+        {
+            if (throwIfNotFound)
+            {
+
+                using (var webRequest = (HttpWebRequest)WebRequest.Create(new Uri(this.baseAddress.OriginalString + requestUri)))
+                {
+                    webRequest.Method = "GET";
+
+                    // add authorization header
+                    webRequest.Headers.Add("Authorization", this.authenticationHeaderProvider.GetAuthorizationHeader());
+
+                    // add custom headers
+                    AddCustomHeaders(webRequest, customHeaders);
 
+                    // perform request and get response
+                    // don't close the WebResponse here because we may need to read the response stream later 
+                    var webResponse = webRequest.GetResponse() as HttpWebResponse;
+
+                    // message received
+                    return ReadResponseMessageAsync(webResponse);
+                }
+            }
+            else
+            {
+                //this.ExecuteAsync(
```

Comment:
looking at it with fresh eyes: I think the intention was rather to have  try { //code } catch(Exception) { if (throwIfNotFound) throw; }. otherwise if throwifNotFound is false, the request is not even processed.